code/API_definitions/qod-provision.yaml

```diff
@@ -0,0 +1,876 @@
+      responses:
+        "200":
+          description: Returns information about an active provision
+          headers:
+            x-correlator:
+              $ref: "#/components/headers/x-correlator"
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ProvisionInfo"
+              examples:
+                PROVISION_AVAILABLE:
+                  $ref: "#/components/examples/PROVISION_AVAILABLE"
+                PROVISION_UNAVAILABLE:
+                  $ref: "#/components/examples/PROVISION_UNAVAILABLE"
+        "400":
+          $ref: "#/components/responses/Generic400"
+        "401":
+          $ref: "#/components/responses/Generic401"
+        "403":
+          $ref: "#/components/responses/Generic403"
+        "404":
+          $ref: "#/components/responses/Generic404"
+        "500":
+          $ref: "#/components/responses/Generic500"
+        "503":
+          $ref: "#/components/responses/Generic503"
+      security:
+        - openId:
+            - "qod-provision:provisions:read"
+
+    delete:
+      tags:
+        - QoD Provision
+      summary: Deletes a QoD provision
+      description: |
+        Release resources related to QoS provision.
+
+        If the notification callback is provided and the provision status was `AVAILABLE`, when the deletion is completed, the client will receive in addition to the response a `PROVISION_STATUS_CHANGED` event with
+        - `status` as `UNAVAILABLE` and
+        - `statusInfo` as `DELETE_REQUESTED`
+        There will be no notification event if the `status` was already `UNAVAILABLE`.
+      operationId: deleteProvision
+      parameters:
+        - $ref: "#/components/parameters/provisionId"
+        - $ref: "#/components/parameters/x-correlator"
+      responses:
+        "204":
+          description: Provision deleted
+          headers:
+            x-correlator:
+              $ref: "#/components/headers/x-correlator"
+        "202":
+          description: Deletion request accepted to be processed. It applies for an async deletion process. `status` in the response will be `AVAILABLE` with `statusInfo` set to `DELETE_REQUESTED`.
+          headers:
+            x-correlator:
+              $ref: "#/components/headers/x-correlator"
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ProvisionInfo"
+        "400":
+          $ref: "#/components/responses/Generic400"
+        "401":
+          $ref: "#/components/responses/Generic401"
+        "403":
+          $ref: "#/components/responses/Generic403"
+        "404":
+          $ref: "#/components/responses/Generic404"
+        "500":
+          $ref: "#/components/responses/Generic500"
+        "503":
+          $ref: "#/components/responses/Generic503"
+      security:
+        - openId:
+            - "qod-provision:provisions:delete"
+
+  /provisions/retrieve:

```

Comment:
Would something like `/provisions/active` be a more intuitive to get the active provision on the current device.

Reply:
I see that the description is wrong as it returns all provisions regardless the status. I'll have to fix that. And as it is a POST modeled to get information, the path is usually modeled as an action. A POST /resource/active would be seen as an endpoint to create a resource in active status.